### ISSUE TITLE
Add com.sun:ldapbp:1.0

### ISSUE THREAD

**Issue by rwinch** [spring-io/platform#196]
Somehow Spring LDAP's `com.sun:ldapbp:1.0` has not been mapped in the Spring IO versions. This was recognized when I updated to the latest Gradle version. The minimum changes to demonstrate the issue can be found [here](https://github.com/spring-projects/spring-ldap/compare/rb2.0.2.RELEASE...rwinch:springio?expand=1)

**NOTE** The the actual unit tests do not pass due to dependency issues that have been fixed in master, but this is the minimum changes required to reproduce this particular issue.

To see the issue use the rb2.0.2.RELEASE branch and run:

   gradlew springIoCheck

Everything passes. Apply the changes on the springio branch provided above and run:

```
$ gradlew springIoCheck
Attributes found in options. Existing attributes will be replaced due to assignment. Please use 'attributes' method instead as current behaviour will be removed in future
:spring-ldap-core:springIoAlternativeDependenciesCheck
:spring-ldap-core:springIoDependencyVersionMappingCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':spring-ldap-core:springIoDependencyVersionMappingCheck'.
> The following direct dependencies do not have Spring IO versions: 
      - com.sun:ldapbp
  Please refer to the plugin's README for further instructions: https://github.com/spring-projects/gradle-plugins/tree/master/spring-io-plugin#dealing-with-unmapped-dependencies

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.095 secs
```

**NOTE** This may mean there is an issue with Spring IO plugin too

**Comment by wilkinsona** [comment]
Interesting discovery. Thanks, Rob.

The change in the plugin's behaviour is due to [this change](https://github.com/spring-projects/gradle-plugins/commit/b03d091df578c45f1f8e7870bd903ecde603d953) in the propdeps plugin. All that's necessary to cause the failure is to check out the rb2.0.2.RELEASE branch and change the version of the propdeps plugin to 0.0.7.

By default, the Spring IO plugin analyses dependencies in the `runtime` configuration. In 0.0.6 the propdeps plugin [was changed](https://github.com/spring-projects/gradle-plugins/commit/b03d091df578c45f1f8e7870bd903ecde603d953) so that the `compile` configuration now extends the `provided` and `optional` configurations, i.e. the `runtime` configuration, which inherits from compile, now contains everything in the `provided` and `optional` configurations when previously it did not. This means that the dependency on `com.sun:ldapbp` in the `provided` configuration is now visible to the Spring IO plugin and the build fails as it's not part of the Platform.

**Comment by wilkinsona** [comment]
@rwinch I've added the dependency in 1.0.5 and merged forwards into 1.1.x and master. That means that, ideally, there'd be a Spring LDAP 2.0.3.RELEASE before 1.0.5 is released (in the next month or so).

**Comment by rwinch** [comment]
@wilkinsona Thanks for the fast turnaround. I'm baffled by how it was just the popdeps update (as I thought I had tried that). I plan to have a Spring LDAP release within the next week, so that should be well within the Spring IO release.

Thanks again for digging in and the fix.
